shopelectro/views/catalog.py

```diff
@@ -25,6 +26,62 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
+def get_view_type(request):
+    view_type = request.session.get('view_type', 'tile')
+    assert view_type in ['list', 'tile']
+    return view_type
+
+
+def get_catalog_context(request, category, raw_tags, page_number, per_page, sorting_index):
+    all_tags = newcontext.Tags(models.Tag.objects.all())
+
+    if raw_tags:
+        selected_tags = newcontext.tags.Checked404Tags(
+            newcontext.tags.ParsedTags(
+                tags=all_tags,
+                raw_tags=raw_tags,
+            )
+        )
+    else:
+        selected_tags = newcontext.Tags(models.Tag.objects.none())
+
+    products = newcontext.products.OrderedProducts(
+        products=newcontext.products.TaggedProducts(
+            products=newcontext.products.ProductsByCategory(
+                products=newcontext.products.ActiveProducts(
+                    newcontext.Products(
+                        models.Product.objects.all(),
+                    ),
+                ),
+                category=category,
+            ),
+            tags=selected_tags,
```

Comment:
it'll be more clear to move short args to the first place.
This example looks more clear, i think:
```
products = context...(
    tags=my_tags_set,
    products=context...(
        products=context...(
            products=context...(
                products=context...(
            )
         )
      )
   )
```